### DISCUSSION TITLE
Smatra has problem with Chinese characters and it causes searching problem.

### DISCUSSION THREAD

**Discussion by balmydrizzle** [sumatrapdfreader/sumatrapdf#3143] [Q&A] [answered]
I find when I type Chinese to search in Sumatra. It will convert to a different-code but similar Chinese, which cause searching failed. But it's ok in other pdf readers. Why?

For example, I typed in '木' (unicode = E6 9C A8), Sumatra will convert it to another look-similar Chinese '⽊' (unicode = E2 BD 8A), which cause searching failed. What's wrong with Sumatra when dealing with Chinese character typed-in? It seems that Sumatra change another code-page and disobey the original code-page of the character I typed-in.

Mostly CJK characters reside in and are rom cjk unified ideographs, but Smatra seems to convert them to those under 'CJK Radicals Supplement'. This cause above issue. 

Most PDF reader deal this well. How to fix this in Sumatra?

P.S. I live in Taiwan and use Traditional Chinese. But that latter Chinese character looks like it belongs to different code page.

**Comment by GitHubRulesOK** [comment, marked as answer]
one problem is PDF does not use Unicode so when written and converted to PDF it will often not have the source data, so without a sample its difficult to guess why, however, SumatraPDF only shows what's been detected by MuPDF which in turn uses a font lookup library for characters like A B C "Tree" ⽊ so CJK is more variable in detection, but agreed if its seen by some libraries as different that should also be the case with MuPDF
```
木 | U+6728 | CJK UNIFIED IDEOGRAPH (U+6728) | 99 | Han
⽊ | U+2F4A | KANGXI RADICAL TREE            | 34 | Han
```

**Reply by balmydrizzle** [reply]
I see, thanks! CJK is always a dilemma for UNICODE.

**Comment by GitHubRulesOK** [comment]
By way of explanation to others Sumatra PDF will only find characters in the search box as "literal" plain text so the two symbols here need separate search even if seen as similar to the eye.  Using print to pdf this page then open in SumatraPDF we can text in and out so if the inputs are different the search needs to be different (common request is why cannot SumatraPDF treat e and accented e's as identical.)

Copy and pasted from Left source viwed above centre in Sumatra PDF (pasted into find box) and as the extracted on right shows both characters are distinct.
![image](https://user-images.githubusercontent.com/12754913/204631779-23ba19ec-172c-4fe9-9d57-c09d6996b7f3.png)
